metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -156,278 +162,416 @@ def ensure_field_level_settings_are_normalized(
             )
             if columns_being_dropped:
-                report.report_dropped(
-                    f"The max_number_of_fields_to_profile={config.max_number_of_fields_to_profile} reached. Profile of columns {dataset_name}({', '.join(sorted(columns_being_dropped))})"
+                self.report.report_dropped(
+                    f"The max_number_of_fields_to_profile={self.config.max_number_of_fields_to_profile} reached. Profile of columns {self.dataset_name}({', '.join(sorted(columns_being_dropped))})"
                 )
         return columns_to_profile
 
-    @classmethod
+    @_run_with_query_combiner
+    def _get_column_type(self, column_spec: _SingleColumnSpec, column: str) -> None:
+        column_spec.type_ = BasicDatasetProfilerBase._get_column_type(
+            self.dataset, column
+        )
+
+    @_run_with_query_combiner
+    def _get_column_cardinality(
+        self, column_spec: _SingleColumnSpec, column: str
+    ) -> None:
+        nonnull_count = self.dataset.get_column_nonnull_count(column)
+        column_spec.nonnull_count = nonnull_count
+
+        unique_count = None
+        try:
+            unique_count = self.dataset.get_column_unique_count(column)
+            pct_unique = float(unique_count) / nonnull_count
+        except Exception:
+            logger.exception(
+                f"Failed to get unique count for column {self.dataset_name}.{column}"
+            )
+        column_spec.unique_count = unique_count
+
+        # Logic adopted from Great Expectations.
+        cardinality = None
+        if unique_count is None or unique_count == 0 or pct_unique is None:
+            cardinality = OrderedProfilerCardinality.NONE
+        elif pct_unique == 1.0:
+            cardinality = OrderedProfilerCardinality.UNIQUE
+        elif pct_unique > 0.1:
+            cardinality = OrderedProfilerCardinality.VERY_MANY
+        elif pct_unique > 0.02:
+            cardinality = OrderedProfilerCardinality.MANY
+        else:
+            if unique_count == 1:
+                cardinality = OrderedProfilerCardinality.ONE
+            elif unique_count == 2:
+                cardinality = OrderedProfilerCardinality.TWO
+            elif unique_count < 60:
+                cardinality = OrderedProfilerCardinality.VERY_FEW
+            elif unique_count < 1000:
+                cardinality = OrderedProfilerCardinality.FEW
+            else:
+                cardinality = OrderedProfilerCardinality.MANY
+
+        column_spec.cardinality = cardinality
+
+    @_run_with_query_combiner
+    def _get_dataset_rows(self, dataset_profile: DatasetProfileClass) -> None:
+        dataset_profile.rowCount = self.dataset.get_row_count()
+
+    @_run_with_query_combiner
+    def _get_dataset_column_min(
+        self, column_profile: DatasetFieldProfileClass, column: str
+    ) -> None:
+        if self.config.include_field_min_value:
+            column_profile.min = str(self.dataset.get_column_min(column))
```

Comment:
The min/max are also used for columns with strings

We decided that instead of doing a union of int, float, string, etc, it'd be easier/clearer to simply put strings into the model